Add AddQuestionPage and update add_question logics to support translations

ployniti opened the pull request:
# Describe your changes
Add a `AddQuestionPage.js` in frontend to replace `add_question_form.html` in backend.
The `AddQuestionPage` supports getting translations for questions, answer choices, and the word "other" when user tries to add a new question.
The changes also update the logic to store the new question and its translations into the database.

There is no button on the `LandingPage.js` to access this page yet since we're updating the UI for that but the changes can be tested by going to `http://localhost:3000/addQuestion`.

# Issue ticket number and link
https://github.com/users/smithev95/projects/1/views/1?pane=issue&itemId=69712224

# Testing done
Local UI testing.

# Checklist before requesting a review
[x] I have performed a self-review of my code
[ ] If it is a core feature, I have added thorough tests.

hsean reviewed (changes requested):
Could you please add "Translators" to requirements.txt.  After that I will approve the pull request.  

Translations can be a bit finicky.  Sometimes I get the error below.

![translate_question_error](https://github.com/smithev95/NEFP-Capstone/assets/9604486/d75c0d6b-7a66-4676-ab4c-d4a5229a2fc7)

ployniti commented:
> Could you please add "Translators" to requirements.txt. After that I will approve the pull request.
> 
> Translations can be a bit finicky. Sometimes I get the error below.
> 
> ![translate_question_error](https://private-user-images.githubusercontent.com/9604486/346260994-d75c0d6b-7a66-4676-ab4c-d4a5229a2fc7.PNG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAyNzU4NDUsIm5iZiI6MTcyMDI3NTU0NSwicGF0aCI6Ii85NjA0NDg2LzM0NjI2MDk5NC1kNzVjMGQ2Yi03YTY2LTQ2NzYtYWI0Yy1kNGE1MjI5YTJmYzcuUE5HP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcwNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MDZUMTQxOTA1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NDk4YTQyYWIwOTUxMGUzNGQ3OGQwZGQ3YmJlODU3YTMwMTJlZDllY2Y5ZGRhZDBhNGFkNzE0ZTVjZDRkZWI3NSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.wmOKNeH5YjO3BPCRSLAng5WoV_VQWlXJTfhbx2F_qyY)

Added `translators` to `requirements.txt`.
I also updated my get request and the url to get translations a bit to help with the above issue.

ployniti commented:
Sorry y'all, I tried to pull the latest changes from `main` into this branch so I can add `addQuestionPage` to the `navbar` that was just merged from Sean's `tk-58` branch. Hopefully this didn't make things too difficult for others.

hsean reviewed (approved):
Everything works on my local machine.  Merging request.
